test/jdk/java/lang/invoke/lambda/TestStableLambdaName.java

```diff
@@ -0,0 +1,301 @@
+                                altInterface, altMethod);
+                        lambdaNames.add(removeHashFromLambdaName(lambda.getClass().getName()));
+                        numOfCreatedLambdas++;
+                    }
+                }
+                lambda = lambdaWithMultipleAltInterfaceAndMultipleAltMethods(interfaceMethod, interfaceClass, flags);
+                lambdaNames.add(removeHashFromLambdaName(lambda.getClass().getName()));
+                numOfCreatedLambdas++;
+            }
+        }
+    }
+
+    private static void createLambdasWithDifferentParameters(Set<String> lambdaNames) throws Throwable {
+        // All lambdas with flags 0
+        createPlainLambdas(lambdaNames, lambdaType.NOT_SERIALIZABLE_NO_ALT_METHODS_NO_ALT_INTERFACES.index);
+
+        // All lambdas with flags 1
+        createPlainLambdas(lambdaNames, lambdaType.SERIALIZABLE_ONLY.index);
+
+        // All lambdas with flags 2
+        createLambdasWithAltInterfaces(lambdaNames, lambdaType.NOT_SERIALIZABLE_HAS_ALT_INTERFACES.index);
+
+        // All lambdas with flags 3
+        createLambdasWithAltInterfaces(lambdaNames, lambdaType.SERIALIZABLE_HAS_ALT_INTERFACES.index);
+
+        // All lambdas with flags 4
+        createLambdasWithAltMethods(lambdaNames, lambdaType.NOT_SERIALIZABLE_HAS_ALT_METHODS.index);
+
+        // All lambdas with flags 5
+        createLambdasWithAltMethods(lambdaNames, lambdaType.SERIALIZABLE_HAS_ALT_METHODS.index);
+
+        // All lambdas with flags 6
+        createLambdasWithAltInterfacesAndAltMethods(lambdaNames, lambdaType.NOT_SERIALIZABLE_HAS_ALT_METHODS_HAS_ALT_INTERFACES.index);
+
+        // All lambdas with flags 7
+        createLambdasWithAltInterfacesAndAltMethods(lambdaNames, lambdaType.SERIALIZABLE_HAS_ALT_METHODS_HAS_ALT_INTERFACES.index);
+    }
+
+    public static void main(String[] args) throws Throwable {
+        numOfCreatedLambdas = 0;
+        MethodType methodTypeForMethod1 = methodTypes[0];
+        MethodType methodTypeForMethod2 = methodTypes[1];
+        methodHandles = new MethodHandle[]{lookup.findStatic(TestStableLambdaName.class, "method1", methodTypeForMethod1),
+                lookup.findStatic(TestStableLambdaName.class, "method2", methodTypeForMethod2)};
+
+        Set<String> lambdaClassStableNames = new HashSet<>();
+        createLambdasWithDifferentParameters(lambdaClassStableNames);
+
+        Set<String> lambdaClassStableNamesTest = new HashSet<>();
+        createLambdasWithDifferentParameters(lambdaClassStableNamesTest);
+
+        if (lambdaClassStableNames.size() != lambdaClassStableNamesTest.size()) {
+            throw new RuntimeException("Same number of different names must be created in the name creation run and the test run");
+        }
+
+        if (!lambdaClassStableNamesTest.containsAll(lambdaClassStableNames)) {
+            throw new RuntimeException("Same stable names must be created in the name creation run and the test run");
```

Comment:
List the difference in these error messgaes.

Reply:
One is about number of different stable names, and the other is about stable name values that have to be the same. Should this be more distinctive?

Reply:
If we want to print message `Expected name for class is Lambda$hashCode but created name is Lambda$otherHashCode`, we would need to keep track of the names that should be the same. First created lambda in the first run with the first created lambda in the second run etc. It is expected to have all the different names if we do not have this feature we are implementing, so message would cover 1/1400 cases for different names. Should it be as it is now (after the change) or should I use list and keep track of the corresponding names?